### PR TITLE
Add lints

### PR DESCRIPTION
**Pull request by domesticmouse**
Adding lints and running `dart fix` across embedded projects

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Comment by domesticmouse** [comment]
> I'm going to loop through and put in the fixes for the use_super_parameters lint!

I've added a bunch more changes from `dart fix` and friends. PTAL +)

File: examples/cookbook/design/package_fonts/analysis_options.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-include: package:flutter_lints/flutter.yaml
+# Take our settings from the example directory's main analysis_options.yaml
+# file. If necessary for a particular example, this file can also include
+# overrides for individual lints.
 
-linter:
-  rules:
-    prefer_single_quotes: true
+include: ../../../analysis_options.yaml
```
Comment (khanhnwin) [inline examples/cookbook/design/package_fonts/analysis_options.yaml:5]: nit: should there be a newline at the end of this file?

File: examples/cookbook/effects/download_button/analysis_options.yaml [hunk 2]
```diff
@@ -1,6 +1,5 @@
-include: package:flutter_lints/flutter.yaml
+# Take our settings from the example directory's main analysis_options.yaml
+# file. If necessary for a particular example, this file can also include
+# overrides for individual lints.
 
-linter:
-  rules:
-    avoid_print: true
-    prefer_single_quotes: true
+include: ../../../analysis_options.yaml
```
Comment (khanhnwin) [inline examples/cookbook/effects/download_button/analysis_options.yaml:5]: same nit: newline at the end of this file?

File: examples/cookbook/effects/nested_nav/analysis_options.yaml [hunk 3]
```diff
@@ -1,6 +1,5 @@
-include: package:flutter_lints/flutter.yaml
+# Take our settings from the example directory's main analysis_options.yaml
+# file. If necessary for a particular example, this file can also include
+# overrides for individual lints.
 
-linter:
-  rules:
-    avoid_print: true
-    prefer_single_quotes: true
+include: ../../../analysis_options.yaml
```
Comment (khanhnwin) [inline examples/cookbook/effects/nested_nav/analysis_options.yaml:5]: Looks like all of the analysis_options.yaml imports are missing newline at the end of the file

Comment (domesticmouse) [reply examples/cookbook/effects/nested_nav/analysis_options.yaml:5]: And there is only a hundred of so of 'em =)

Comment (domesticmouse) [reply examples/cookbook/effects/nested_nav/analysis_options.yaml]: On the upside, only 31 or so had trailing newlines.

**Review by sfshaza2 — APPROVED** [review]
Oh, @domesticmouse, what a herculean task. LGTM, assuming the build passes.

**Review by khanhnwin — APPROVED** [review]

**Comment by khanhnwin** [comment]
^ What Shams said. Thank you @domesticmouse!

**Comment by khanhnwin** [comment]
Woohoo! It's green!

**Comment by domesticmouse** [comment]
> Woohoo! It's green!

And it's landed =)

**Pull request merged**